src/Options/Applicative/Types.hs

```diff
@@ -242,14 +242,14 @@ data OptReader a
   -- ^ flag reader
   | ArgReader (CReader a)
   -- ^ argument reader
-  | CmdReader (Maybe String) [String] (String -> Maybe (ParserInfo a))
+  | CmdReader (Maybe String) [(String, ParserInfo a)]
```

Comment:
This is the main change.

The `lookup` function is now not embedded creating this type, but selected in the searchParser.